docs/operate-and-deploy/migrations-tool.md

```diff
@@ -0,0 +1,609 @@
+following command creates a new migrations project in the local 
+`./my/migrations/dir` directory to connect to a ksqlDB server listening at 
+`http://localhost:8088` (which is accessed from within the docker container 
+at `http://host.docker.internal:8088`):
+
+```
+docker run -v $PWD/my/migrations/dir:/share/ksql-migrations confluentinc/ksqldb-server:{{ site.ksqldbversion }} ksql-migrations new-project /share/ksql-migrations http://host.docker.internal:8088
+```
+
+Similarly, the following initializes migrations metadata on the ksqlDB server 
+for the same setup:
+
+```
+docker run -v $PWD/my/migrations/dir:/share/ksql-migrations confluentinc/ksqldb-server:{{ site.ksqldbversion }} ksql-migrations --config-file /share/ksql-migrations/ksql-migrations.properties initialize-metadata
+```
+
+See the sections below for more on the different `ksql-migrations` commands.
+
+### Archive / deb / rpm
+
+TODO
+
+Setup and Initialization
+------------------------
+
+### Initial Setup
+
+To get started with the `ksql-migrations` tool, use the `ksql-migrations new-project`
+command to set up the required directory structure and create a config file for
+using the migrations tool. 
+
+```
+ksql-migrations new-project [--] <project-path> <ksql-server-url>
+```
+
+The two required arguments are the path that will be used as the root directory
+for your new migrations project, and your ksqlDB server URL. 
+
+```
+$ ksql-migrations new-project /my/migrations/project/path http://localhost:8088
+
+Creating new migrations project at /my/migrations/project/path
+Creating directory: /my/migrations/project/path
+Creating directory: /my/migrations/project/path/migrations
+Creating file: /my/migrations/project/path/ksql-migrations.properties
+Writing to config file: ksql.server.url=http://localhost:8088
+...
+Migrations project directory created successfully
+Execution time: 0.0080 seconds
+```
+
+This command creates a config file `ksql-migrations.properties`
+in the specified directory, and also creates an empty `/migrations` subdirectory.
+The config file is initialized with the ksqlDB server URL passed as part of the
+command. 
+
+As a convenience, the config file is also initialized with default values 
+for other [migrations tool configurations](#config-reference) commented out.
+These additional, optional configurations include configs required to access
+secure ksqlDB servers, such as credentials for HTTP basic authentication or TLS keystores
+and truststores, as well as optional configurations specific to the migrations tool.
+
+See the [config reference](#config-reference) for details on individual configs.
+See [here](#connecting-to-confluent-cloud-ksqldb) for the configs required to 
+connect to a Confluent Cloud ksqlDB cluster.
+
+### Initialize Migrations Metadata
+
+The `ksql-migrations` tool keeps track of applied migration versions in a ksqlDB
+stream and table, the *migrations metadata* stream and table.
```

Comment:
```suggestion
stream and table, named the *migrations metadata* stream and table.
```

Reply:
I'm not a huge fan of this suggestion since the name of a stream/table refers to something else -- the (default) name of the migrations metadata stream is `MIGRATION_EVENTS` and the (default) name of the migrations metadata table is `MIGRATION_SCHEMA_VERSIONS`.

I've left the original text for now but can update to say "called the migrations metadata stream and table" if you prefer?

Reply:
That's cool, maybe mention the default names?

Reply:
Leaving this as is for now -- I don't think we need to mention the names in this sentence since they're shown in the example output a few sentences below, and the defaults are also in the config reference.